src/viewer/PdfJsViewer.jsx

```diff
@@ -0,0 +1,167 @@
+import React, { Component } from 'react'
+import { Document, Page } from 'react-pdf/dist/entry.webpack'
+import cx from 'classnames'
+import throttle from 'lodash/throttle'
+import { Button, Spinner } from 'cozy-ui/react'
+import withFileUrl from './withFileUrl'
+import NoViewer from './NoViewer'
+import styles from './styles'
+
+export const MIN_SCALE = 0.25
+export const MAX_SCALE = 3
+
+const ToolbarButton = ({ icon, onClick, disabled }) => (
+  <Button
+    iconOnly
+    subtle
+    theme="secondary"
+    className="u-p-half u-m-half"
+    icon={icon}
+    onClick={onClick}
+    disabled={disabled}
+  />
+)
+
+export class PdfJsViewer extends Component {
+  state = {
+    totalPages: 1,
+    scale: 1,
+    currentPage: 1,
+    loaded: false,
+    errored: false,
+    width: null
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.wrapper = null
+    this.resizeListener = null
+
+    this.onLoadSuccess = this.onLoadSuccess.bind(this)
+    this.onLoadError = this.onLoadError.bind(this)
+    this.nextPage = this.nextPage.bind(this)
+    this.previousPage = this.previousPage.bind(this)
+    this.scaleUp = this.scaleUp.bind(this)
+    this.scaleDown = this.scaleDown.bind(this)
+    this.setWrapperSize = this.setWrapperSize.bind(this)
```

Comment:
you could remove all these binds if you used class properties instead of methods like you already do for `setWrapperSize`

Reply:
you can even remove the constructor if you define `wrapper` and `resizeListener` as properties also